refactored mbo

chrstngyn opened the pull request:
## Description
Refactored MBO control to include clock, CTC, TrainModel, TrackController dependencies.

## Changes
Import statements for clock, CTC, TrainModel, TrackController.

## Testing
n/a

## Additional Information
n/a

Commit:
refactored mbo

michaelplazek commented on MBO/src/mbo/model/MovingBlockOverlay.java:
```diff
@@ -1,24 +1,47 @@
   // MBO constructor
   private MovingBlockOverlay() {
-    schedule = new Schedule();
+    Schedule schedule = new Schedule();
```
You want this to be an instance variable

michaelplazek commented on MBO/src/mbo/model/MovingBlockOverlay.java:
```diff
@@ -29,15 +52,15 @@ private MovingBlockOverlay() {
   * */
 
-  public static MovingBlockOverlay getInstance() {
+  public MovingBlockOverlay getInstance() {
```
this should be static

michaelplazek commented on MBO/src/mbo/view/MovingBlockOverlayController.java:
```diff
@@ -93,24 +98,23 @@
 
   @Override
   public void initialize(URL location, ResourceBundle resources) {
+    setMBOMode(mboMode);
```
this should initialize as false because we're starting fixed-block mode, ie. setMboMode(false);

michaelplazek commented on MBO/src/mbo/view/MovingBlockOverlayUserInterface.java:
```diff
@@ -6,19 +6,19 @@
 import javafx.scene.Scene;
 import javafx.stage.Stage;
 
+@SuppressWarnings("ALL")
```
You shouldnt need this

michaelplazek reviewed (changes requested):
Overall, just some small changes before we merge. You also need to run checkstyle.

Commit:
made changes requested

Commit:
checkstyle fixes

Commit:
removed getInstance() from MovingBlockOverlayInterface.java

Commit:
removed controller statement from mbo.fxml

Commit:
more checkstyle fixes

michaelplazek commented on MBO/src/mbo/model/ExportSchedule.java:
```diff
@@ -0,0 +1,97 @@
+          try {
+            br = new BufferedReader(new FileReader(new File(RED_LINE)));
+          } catch (FileNotFoundException e) {
+            System.out.println("File not found!");
```
What is the purpose of this?

chrstngyn replied:
check if redLine file exists

michaelplazek reviewed (changes requested):
Still getting some checkstyle errors from ExportSchedule

Jzang2011 committed:
fixed checkstyle errors

- fixed check style errors for Christine so I can make this fucking
class diagram.

Commit:
endless checkstyle fixes

Commit:
Merge branch 'refactor-mbo' of https://github.com/michaelplazek/on-track into refactor-mbo

Jzang2011 committed:
fixed merge conflict
